client/signup/steps/design-picker/index.jsx

```diff
@@ -119,6 +120,11 @@ export default function DesignPickerStep( props ) {
 		sort: sortBlogToTop,
 	} );
 
+	useEffect( () => {
+		// Sets the category selection which is important for the difm flow
+		dispatch( selectDesignCategory( categorization.selection ) );
+	}, [ categorization.selection, dispatch ] );
+
```

Comment:
Each time a category changes we set the redux state in the front end state. Thought of including it here since,

- externalising this as a dependency  will reduce cohesion of where this should actually happen. And this globally required state needs to live in the global redux state to be used anywhere.
- the design picker might be moved out of the signup framework soon.

This state is then persisted as a blog option on site creation.

https://github.com/Automattic/wp-calypso/blob/3d314a83071c01a4168e8285448c83b2b95a3f85/client/lib/signup/step-actions/index.js#L137

Reply:
You're right it is a bit strange having this state duplicated in two places. You could maybe submit the selected category as a `signupDependency` instead? That way it's just one other piece of data this step provides, and it wouldn't be unexpected for the UI to maintain it's only transient state for the category selection before submitting the step.

You could also call `selectDesignCategory()` in the `pickDesign()` method if you wanted to keep it out of `signupDependencies`.

Reply:
I have taken your advice and switched to signup dependencies! 👍